### DISCUSSION TITLE
How to create and register a UDF to be used by any pipeline

### DISCUSSION THREAD

**Discussion by dadepo** [ArroyoSystems/arroyo#214] [Q&A] [answered]
I am interested in being able to define a UDF once, register it, and have it useable across the system.

Looking at the documentation here https://doc.arroyo.dev/api-reference/pipelines/create-pipeline I see it is possible to register a UDF as part of a pipeline.

What I want to be able to do is to register the UDF, independent of a particular pipeline, in other to make useable from any/every pipeline.

Is this currently possible?

**Comment by mwylde** [comment, marked as answer]
Hey @dadepo apologies for the slow response, we missed this post.

Currently there's no support for "stored" UDFs. That would be a great feature though and would fit in nicely with our existing support for storing connection tables. I've created an issue to track this: https://github.com/ArroyoSystems/arroyo/issues/280.
